include/tvm/relax/nested_msg.h

```diff
@@ -305,6 +305,101 @@ NestedMsg<T> MapToNestedMsg(StructInfo sinfo, FType fmapleaf) {
   }
 }
 
+/*!
+ * \brief Map expr with possible nested-tuple to nested message,
+ * according to its struct info.
+ *
+ * This function will unpack recursive sinfo, run fstepproces in every
+ * recursion step, and run fmapleaf for each leaf.
+ * then recursively combines the results together into a NestedMsg.
+ *
+ * The nesting structure will corresponds to the tuple structure.
+ *
+ * \param expr The input expression with tuple struct info.
+ * \param fstepprocess The processing function for each step with signature
+ *             Expr fstepprocess(Expr, size_t)
+ * \param fmapleaf The mapping function for each leaf with signature
+ *             NestedMsg<T> fmap(Expr)
+ * \tparam T the content type of nested msg
+ * \tparam FType The mapping function type
+ */
+template <typename T, typename FType1, typename FType2>
+NestedMsg<T> MapToNestedMsg(Expr expr, FType1 fstepprocess, FType2 fmapleaf) {
```

Comment:
It looks to me that the only use of `fstepprocess` is to retrieve the field of Tuple, and I think it makes sense to make it the default behavior (when we recurse into sinfo along with the expr) and doesn't have to pass it in. Just like the TransformTupleLeaf below.

Reply:
Makes sense. Let's add a bool flag arg in `NestedMsg<T> MapToNestedMsg(Expr expr, FType fmapleaf)` to decide whether recurse by itself or its info

Reply:
Also remove this and `MapToNestedMsgNew`

Reply:
Or we can rename `MapToNestedMsgNew` to `MapToNestedMsgBySInfo` and use it